rfcs/text/0005-host-metric-fields.md

```diff
@@ -26,6 +26,17 @@ Proposed 7 new fields are:
 <!--
 Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
 -->
+This RFC calls for the addition of host fields to collect basic monitoring metrics from a host or VM such as CPU, network and disk.
+
+| field | type | description |
+| --- | --- | --- |
+| `host.cpu.pct` | scaled_float | Percent CPU used. This value is normalized by the number of CPU cores and it ranges from 0 to 1. |
+| `host.network.in.bytes` | long | The number of bytes received on all network interfaces by the host in a given period of time. |
```

Comment:
In the observer namespaces, ECS uses "ingress" and "egress". For consistency we might want to consider uses those terms.

Reply:
Can we document the nature of this counter. Is it a monotonic counter that is sometimes reset (e.g. system restart)?

Reply:
@cyrille-leclerc These values will actually be gauges. For example, `host.network.in.bytes` will be the total bytes received aggregated among all network interfaces in 10 seconds. Next collection period, this value will change, might go up or down depends on the network usage of the next 10 seconds.

Reply:
++ on aligning on `ingress` and `egress`

Reply:
thanks @kaiyan-sheng I was used to ever increasing counter this kind of metric but I don't know what's the state of the art today. 
One benefit of ever increasing counters being that the collector can miss some collects without losing data.

Reply:
I think we're confusing two different things here.
The main purpose for this set of host metrics is to align the different resource providers, such as public cloud infrastructures, physical hosts, VMware clusters, other hypervisors, i.e. bring hosts/VMs to a common denominator. The purpose is not to capture high fidelity system metrics because some providers (AWS, GCP, Azure, etc.) simply don't provide these. 
Granted, the Metricbeat system module can do much more than just the proposed CPU metrics and should be used wherever possible. 
Be making these metrics part of ECS we want improve user experience, i.e. we can always show basic health information about any hosts anywhere, and recommend users to enable the system module for enhanced details.

Reply:
@webmat @cyrille-leclerc Agree on the benefit of ever-increasing counters. We will definitely keep all the counters as they are right now from the system module. We only added extra calculation using these counters to get gauges in system module to match whatever we get from other resource providers such as AWS, Azure, and GCP. Unfortunately, they don't provide ever-increasing counters in their monitoring metrics as @sorantis mentioned above.

Also in the UI side, it's definitely easier to have metrics as gauges than counters.